### PR TITLE
perf(orchestrator): replace FetchCandidateIssues scan with FetchIssueByID in HandleRetryTimer

### LINKED ISSUE #206: Optimize retry timer validation to single-issue fetch
`HandleRetryTimer` calls `FetchCandidateIssues` (full paginated sweep of all active issues) to validate a single retried issue. It then performs a linear scan via `findIssueByID` to locate the issue in the result set. At scale (hundreds of active issues, multiple concurrent retry timers), this produces O(pages) tracker API calls per timer event when O(1) would suffice.

Replace the `FetchCandidateIssues` → `findIssueByID` sequence with `FetchIssueByID` → local active-state check against `params.ActiveStates`. This reduces each retry timer to a single API call regardless of the total active issue count.

The equivalence between candidate-set membership and `FetchIssueByID` + active-state check must be verified: `FetchCandidateIssues` applies tracker-level filtering (JQL query, label filter, project scoping) that `FetchIssueByID` does not. If the tracker adapter's candidate query includes constraints beyond state (e.g. issue type exclusion, custom field filter), the optimized path must replicate those constraints or fall back to the full fetch.

Verify: benchmark or integration test confirms single-issue fetch path produces the same eligibility result as the current candidate-set approach for all three tracker adapters (Jira, GitHub, file).

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context

**Type:** Perf

**Intent:** Each retry timer event previously triggered a full paginated sweep of all active issues (`FetchCandidateIssues`) just to validate one issue, costing O(pages) tracker API calls. This replaces that sweep with a single `FetchIssueByID` call plus a local active-state membership check, reducing each timer fire to exactly one API call regardless of scale.

**Related Issues:** #206

### 🧭 Reviewer Guide

**Complexity:** Medium

#### Entry Point

`internal/orchestrator/retry.go` - the core change is in `HandleRetryTimer`, replacing the `FetchCandidateIssues` + `findIssueByID` block (~lines 181-240) with a `FetchIssueByID` call. Three error paths branch from the result: `ErrTrackerNotFound` releases the claim, transient errors reschedule with backoff, and a non-active state releases the claim.

#### Sensitive Areas

- `internal/orchestrator/retry.go`: Retry claim lifecycle - verify the three new branches (not-found, transient error, non-active state) each correctly release or reschedule without leaking claims.
- `internal/orchestrator/retry_test.go`: `mockRetryTracker` is inverted - `FetchCandidateIssues` now panics if called (enforces the new contract at test runtime); `FetchIssueByID` is the primary entry point.
- `internal/scm/github/tracker_test.go`: Covers the PR-as-issue edge case - GitHub adapter returns `ErrTrackerNotFound` for pull requests, so the optimized path correctly releases any PR claim.

### ⚠️ Risk Assessment

- **Breaking Changes:** No breaking changes. `domain.TrackerAdapter` interface is unmodified; `FetchIssueByID` was already present. The `RetryEntry.Error` string changes from `"retry poll failed"` to `"retry issue fetch failed"` for transient fetch errors - any log queries or dashboards matching the old string will need updating.
- **Migrations/State:** No migrations or state changes.

### CODE REVIEW

File: internal/orchestrator/retry.go [hunk 1]
```diff
@@ -208,10 +225,16 @@ func HandleRetryTimer(state *State, issueID string, params HandleRetryTimerParam
...
-	if !found {
-		log.Info("issue no longer active, releasing claim")
+	// Validate the issue's current state against the configured active states.
+	// This is the sole gate replacing the implicit activeStates filter that
+	// FetchCandidateIssues applied at the tracker level. The queryFilter
+	// (adapter-internal JQL or search fragment) is not re-evaluated here:
+	// retry eligibility uses the state constraint only.
+	activeSet := stateSet(params.ActiveStates)
+	if _, active := activeSet[strings.ToLower(issue.State)]; !active {
+		log.Info("issue no longer in active state, releasing claim",
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply internal/orchestrator/retry.go:235]: @copilot This is an intentional and accepted design trade-off documented in both the spec and the code comment at the call site (lines 230-233). The `queryFilter` is a *pickup filter* that determines which issues enter the claim pool on first contact. Once an issue is claimed and a retry is scheduled, the intent is to continue working on it - re-evaluating the queryFilter at retry time serves no correctness purpose. If the queryFilter condition changed (e.g. a label was removed), the next full poll tick will not re-claim the issue, and the retry naturally expires. The only observable impact is at most one extra re-dispatch of an already-claimed issue, which is indistinguishable from normal continuation behavior. Adding a fallback to `FetchCandidateIssues` when a queryFilter is configured would reintroduce the O(pages) cost this PR eliminates, defeating the purpose of the optimization.
